hs_core/hydroshare/utils.py

```
@@ -121,6 +124,143 @@ def group_from_id(grp):
+
+
+def is_federated(homepath):
+    """
+    Get size of a data object from iRODS user zone
+    Args:
+        user: the requesting user
+        fname: the logical iRODS file name with full logical path
+
+    Returns:
+    the size of the file
+    """
+    irods_storage = IrodsStorage('federated')
+    # if HS WWW iRODS proxy user can list homepath, homepath is federated; otherwise, it is not federated
+    return irods_storage.exists(homepath)
+
+
+def get_federated_zone_home_path(filepath):
+    """
+    Args:
+        filepath: the iRODS data object file path that included zone name in the format of /zone_name/home/user_name/file_path
+
+    Returns:
+        the zone name extracted from filepath
+    """
+    if filepath and filepath.startswith('/'):
+        split_path_strs = filepath.split('/')
+        # the Zone name should follow the first slash
+        zone = split_path_strs[1]
+        return '/{zone}/home/{local_proxy_user}'.format(zone=zone, local_proxy_user=settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE)
+    else:
+        return ''
+
+
+def get_fed_zone_file_size(fname):
+    """
+    Get size of a data object from iRODS user zone
+    Args:
+        fname: the logical iRODS file name with full logical path
+
+    Returns:
+    the size of the file
+    """
+
+    irods_storage = IrodsStorage('federated')
+    return irods_storage.size(fname)
+
+
+def get_fed_zone_files(irods_fnames):
+    """
+    Get the file from iRODS federated zone to Django server for metadata extraction on-demand for specific resource types
+    Args:
+        fname: the logical iRODS file name with full logical path
+
+    Returns:
+    the named temp file being copied over to local Django server
+    """
+    ret_file_list = []
+    if isinstance(irods_fnames, basestring):
+        ifnames = string.split(irods_fnames, ',')
+    elif isinstance(irods_fnames, list):
+        ifnames = irods_fnames
+    else:
+        return ret_file_list
```

I think better to raise exception here instead of returning an empty list.